### ISSUE TITLE
Can't build on archlinux

### ISSUE THREAD

**Issue by fdev31** [yoyz/picoloop#2]
While building on archlinux (following debian instructions), on the first non-_clean_ target I get the following errors (it seems the "debianrtaudio" folder shall be populated by empty folders first):

```
rm -f debianrtaudio/RtAudio.o
g++ -I. -MM Machine/MDADrum/profport.cpp -o debianrtaudio/Machine/MDADrum/profport.d
g++ -I. -MM Machine/MDADrum/drumsynth.cpp -o debianrtaudio/Machine/MDADrum/drumsynth.d
g++ -I. -MM Machine/MDADrum/MDADrumUserInterface.cpp -o debianrtaudio/Machine/MDADrum/MDADrumUserInterface.d
cc1plus: fatal error: opening output file debianrtaudio/Machine/MDADrum/MDADrumUserInterface.d: No such file or directory
compilation terminated.
cc1plus: fatal error: opening output file debianrtaudio/Machine/MDADrum/profport.d: No such file or directory
compilation terminated.
cc1plus: fatal error: opening output file debianrtaudio/Machine/MDADrum/drumsynth.d: No such file or directory
compilation terminated.
g++ -I. -MM Machine/MDADrum/MDADrumMachine.cpp -o debianrtaudio/Machine/MDADrum/MDADrumMachine.d
g++ -I. -MM Machine/MidiOutSystem/MidiOutMachine.cpp -o debianrtaudio/Machine/MidiOutSystem/MidiOutMachine.d
g++ -I. -MM Machine/MidiOutSystem/MidiOutUserInterface.cpp -o debianrtaudio/Machine/MidiOutSystem/MidiOutUserInterface.d
cc1plus: fatal error: opening output file debianrtaudio/Machine/MidiOutSystem/MidiOutMachine.d: No such file or directory
compilation terminated.
cc1plus: fatal error: opening output file debianrtaudio/Machine/MDADrum/MDADrumMachine.d: No such file or directory
compilation terminated.
g++ -I. -MM Machine/Twytch/twytch_voice_handler.cpp -o debianrtaudio/Machine/Twytch/twytch_voice_handler.d
g++ -I. -MM Machine/Twytch/twytch_wave.cpp -o debianrtaudio/Machine/Twytch/twytch_wave.d
cc1plus: fatal error: opening output file debianrtaudio/Machine/MidiOutSystem/MidiOutUserInterface.d: No such file or directory
compilation terminated.
g++ -I. -MM Machine/Twytch/twytch_simple_delay.cpp -o debianrtaudio/Machine/Twytch/twytch_simple_delay.d
cc1plus: fatal error: opening output file debianrtaudio/Machine/Twytch/twytch_wave.d: No such file or directory
compilation terminated.
g++ -I. -MM Machine/Twytch/twytch_smooth_filter.cpp -o debianrtaudio/Machine/Twytch/twytch_smooth_filter.d
cc1plus: fatal error: opening output file debianrtaudio/Machine/Twytch/twytch_simple_delay.d: No such file or directory
compilation terminated.
g++ -I. -MM Machine/Twytch/twytch_smooth_value.cpp -o debianrtaudio/Machine/Twytch/twytch_smooth_value.d
cc1plus: fatal error: opening output file debianrtaudio/Machine/Twytch/twytch_voice_handler.d: No such file or directory
compilation terminated.
g++ -I. -MM Machine/Twytch/twytch_step_generator.cpp -o debianrtaudio/Machine/Twytch/twytch_step_generator.d
cc1plus: fatal error: opening output file debianrtaudio/Machine/Twytch/twytch_smooth_filter.d: No such file or directory
compilation terminated.
g++ -I. -MM Machine/Twytch/twytch_stutter.cpp -o debianrtaudio/Machine/Twytch/twytch_stutter.d
cc1plus: fatal error: opening output file debianrtaudio/Machine/Twytch/twytch_smooth_value.d: No such file or directory
compilation terminated.
g++ -I. -MM Machine/Twytch/twytch_switch.cpp -o debianrtaudio/Machine/Twytch/twytch_switch.d
cc1plus: fatal error: opening output file debianrtaudio/Machine/Twytch/twytch_step_generator.d: No such file or directory
compilation terminated.
[...]
```

**Comment by yoyz** [comment]
Thank you for the issue, I have the same here on debian.

I don't know how to : 
- include all the  ".d" file in a target directory for example "debianrtaudio"
- generate all the ".d" file from the ".h" and ".cpp" and put them on "debianrtaudio"
- create this target directory "debianrtaudio" before this include 

The ".d" file is a dependency file which is include in the makefile.
If you have a solution which solve it, with a "if" I take it.
I have the same problem on my debian, and it is really important for me to have different target directory and to be able to build the ".o" without cleaning all the project.

**Comment by fdev31** [comment]
Did you try using a pre-make shell script for instance ?

**Comment by yoyz** [comment]
you mean something like, build.sh which call make ?
Or embedded shell stuff in the makefile.
Maybe it could be the best solution to hide my problem.
But i'm not sure we talk about the same thing.
Could you describe what you think could be the best solution to this problem ?

**Comment by yoyz** [comment]
this problem is in master branch and picoloop-0.67

**Comment by yoyz** [comment]
Should be fixed for debian and raspi1 with tag 0.76c
